pkg/app/redirect/getters.go

```diff
@@ -20,6 +20,24 @@ func (a *appDirector) getRdQuerystringRedirect(req *http.Request) string {
 	)
 }
 
+// getCustomRedirectHeader uses the configured custom header as the redirect
+// path, along with the query parameters from the original request, if present.
+func (a *appDirector) getCustomRedirect(req *http.Request) string {
+	if h := a.customRedirectHeader; h != "" {
+		if r := req.Header.Get(h); r != "" {
+			query := ""
+			if q := req.URL.RawQuery; q != "" {
+				query = "?" + q
+			}
+			return a.validateRedirect(
+				r+query,
+				fmt.Sprintf("Invalid redirect generated from %s header: %%s", h),
+			)
+		}
+	}
+	return ""
```

Comment:
Lets refactor this to get rid of the deeply nested structure. We try to follow the early return / early exit strategy for methods. Furthermore please use descriptive variable names instead of single letters.

Reply:
Done